### PR TITLE
[Engine] Bugfix ArrayOutOfBounds in AbstractScene

### PR DESCRIPTION
**Pull request by KayMD**
Add more documentation, Feb 12 commits resolve Mantis ticket 10711, 10712, 10704, 10705

### CODE REVIEW

File: illaclient/src/main/java/illarion/client/world/interactive/InteractionManager.java [hunk 1]
```diff
@@ -87,24 +87,24 @@ public ItemCount getMovedAmount() {
...
-            draggedObject.dragTo(targetSlot, count);
-        } catch (@Nonnull IndexOutOfBoundsException ex) {
-            LOGGER.error("Tried to drop a item at a container slot that does not exist.", ex);
-        } finally {
-            cancelDragging();
-        }
+                try {
+                    InteractiveContainerSlot targetSlot = itemContainer.getSlot(slot).getInteractive();
+                    draggedObject.dragTo(targetSlot, count); // Enter the contents into the slot
+                } catch (@Nonnull IndexOutOfBoundsException ex) {
```
Comment (mkaring) [inline illaclient/src/main/java/illarion/client/world/interactive/InteractionManager.java]: I consider catching the IndexOutOfBoundsException as very bad practise.
We should rather try to find the reason why the wrong slot is requested.

Comment (KayMD) [inline illaclient/src/main/java/illarion/client/world/interactive/InteractionManager.java]: I have a strong suspicion that the reason is that cancelDragging() was not properly killing the action, which is why I moved that block into an else{} statement.

However, I didn't author the try{} catch{} block that you mention, and have no real idea why it was done in that way to begin with.

Comment (mkaring) [inline illaclient/src/main/java/illarion/client/world/interactive/InteractionManager.java]: Ah. I see. Yes yes.
There is a strong indication here that your IDE is formatting the code.
You might want to check on that.

Comment (KayMD) [inline illaclient/src/main/java/illarion/client/world/interactive/InteractionManager.java]: Can you explain? It looks fine to me. Nothing is wrapping, anyway.
Regardless, is there anything that looks like it needs to be changed, or can this be applied?

Comment (KayMD) [inline illaclient/src/main/java/illarion/client/world/interactive/InteractionManager.java]: I think that might just be caused by viewing it in split. The formatting looks fine to me from the Unified view.

File: illagameengine/src/main/java/org/illarion/engine/graphic/GraphicResolution.java [hunk 2]
```diff
@@ -18,6 +18,10 @@
 import javax.annotation.Nonnull;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
+import java.awt.GraphicsEnviroment;
```
Comment (mkaring) [inline illagameengine/src/main/java/org/illarion/engine/graphic/GraphicResolution.java]: Shouldn't it be java.awt.GraphicsEnviro*n*ment ? ;-)

I fix that.

**Pull request merged**